### PR TITLE
add MigrationGuidance.md

### PR DESCRIPTION
**Pull request by dding3**

### CODE REVIEW

File: MigrationGuidance.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+This guidance is used to provide guidance to BigDL and Analytics zoo users to migrate their existing BigDL/Analytics Zoo applications to use BigDL2.0
+
+* **For BigDL users**
+
+   ***scala application***
+
+   Please follow below table to change the package name of BigDL0.x to BigDL2.0 in your applications
+
+   | BigDL0.x | BigDL2.0  |
```
Comment (jason-dai) [inline MigrationGuidance.md:11]: make it simpler, e.g., 

Change `import com.intel.analytics.bigdl.XYZ` to `import com.intel.analytics.bigdl.dllib.XYZ` except the following:
- ...
- ...
- ....

File: MigrationGuidance.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+   | :---:   | :-: |
+   | com.intel.analytics.bigdl.dataset.xxx | com.intel.analytics.bigdl.dllib.feature.dataset.xxx|
+   | com.intel.analytics.bigdl.transform.xxx | com.intel.analytics.bigdl.dllib.feature.transform.xxx|
+   | com.intel.analytics.bigdl.nn.keras.xxx | com.intel.analytics.bigdl.dllib.keras.xxx|
```
Comment (jason-dai) [inline MigrationGuidance.md:15]: `bigdl.nn.keras` is deprecated and will be removed; will use the Keras APIs in Zoo instead.

File: MigrationGuidance.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+   | com.intel.analytics.bigdl.models.xxx | com.intel.analytics.bigdl.dllib.models.xxx|
+   | com.intel.analytics.bigdl.nn.xxx(xxx != keras) | com.intel.analytics.bigdl.dllib.nn.xxx|
+   | com.intel.analytics.bigdl.optim.xxx | com.intel.analytics.bigdl.dllib.optim.xxx|
+   | com.intel.analytics.bigdl.tensor.xxx | com.intel.analytics.bigdl.dllib.tensor.xxx|
+   | com.intel.analytics.bigdl.visualization.xxx | com.intel.analytics.bigdl.dllib.utils.visualization.xxx|
```
Comment (jason-dai) [inline MigrationGuidance.md:20]: just keep it as `bigdl.dllib.visualization`

**Comment by jason-dai** [comment]
We also need to guide on how to change maven or sbt dependency.

**Comment by dding3** [comment]
Sorry need to close this pr since it includes lots  of unrelated changes and opened a new one https://github.com/intel-analytics/analytics-zoo/pull/4497

**Pull request closed without merging**